python/pyspark/pandas/typedef/typehints.py

```diff
@@ -793,11 +846,23 @@ def extract_types(params: Any) -> Tuple:
   - DataFrame[index_type, [type, ...]]
   - DataFrame[(index_name, index_type), [(name, type), ...]]
   - DataFrame[dtype instance, dtypes instance]
-  - DataFrame[(index_name, index_type), zip(names, types)]\n"""
+  - DataFrame[(index_name, index_type), zip(names, types)]
+  - DataFrame[[index_type, ...], [type, ...]]
+  - DataFrame[[(index_name, index_type), ...], [(name, type), ...]]
+  - DataFrame[dtypes instance, dtypes instance]
+  - DataFrame[zip(index_names, index_types), zip(names, types)]\n"""
             + "However, got %s." % str(origin)
         )
 
 
+def _get_holder(is_index: bool) -> Union[Type[IndexNameTypeHolder], Type[NameTypeHolder]]:
```

Comment:
let's remove this method. It doesn't virtually remove the duplication by dispatching on `is_index`.